### ISSUE TITLE
The controlnet trained by diffusers scripts produce always same result no matter what the input images is

### ISSUE THREAD

**Issue by garyhxfang** [huggingface/diffusers#3120]
### Describe the bug

I train a controlnet with the base model Chilloutmix-Ni and datasets Abhilashvj/vto_hd_train using the train_controlnet.py script provided in diffuses repo
After training I got a controlnet model.
When I inference the image with the model, if I use the same prompt and seed, no matter how I change the control image I used, the pipeline always output the same image as result, which means that the controlnet model doesn't accept the control image as a condition at all.

### Reproduction

The train the controlnet with the scripts in example/controlnet
`accelerate launch train_controlnet.py --pretrained_model_name_or_path="/root/autodl-tmp/chilloutmixckpt" --output_dir="/root/autodl-tmp/mycontrolnet" --dataset_name=Abhilashvj/vto_hd_train --resolution=512 --learning_rate=2e-6 --train_batch_size=1 --gradient_accumulation_steps=4 --num_train_epochs=10 --tracker_project_name="train_controlnet" --checkpointing_steps=10000`

And the code I use for inference is as below
```
from diffusers import StableDiffusionControlNetPipeline, ControlNetModel, UniPCMultistepScheduler,DPMSolverMultistepScheduler
from diffusers.utils import load_image
import torch
base_model_path = "/root/autodl-tmp/chilloutmixckpt"
controlnet_path = "/root/autodl-tmp/mycontrolnet"
controlnet = ControlNetModel.from_pretrained(controlnet_path)
pipe = StableDiffusionControlNetPipeline.from_pretrained(
    base_model_path, controlnet=controlnet
)
# speed up diffusion process with faster scheduler and memory optimization
pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
#pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
control_image = load_image("https://datasets-server.huggingface.co/assets/Abhilashvj/vto_hd_train/--/Abhilashvj--vto_hd_train/train/5/conditioning_image/image.jpg")
control_image.save("./control8.png")
prompt = "1girl, best quality, ultra high res, high quality, ultra-detailed, professional lighting"
negative_prompt = 'paintings, sketches, extremely worst quality, worst quality, extremely low quality, low quality, normal quality, lowres, normal quality, monochrome, grayscale, missing fingers, extra fingers, bad teeth, bad anatomy, bad hands, bad feet, blurry face, bad eyes, slanted eyes, fused eye, skin spots, acnes, skin blemishes, age spot'
# generate image
generator = torch.manual_seed(0)
image = pipe(prompt, num_inference_steps=20, generator=generator, image=control_image).images[0]
image.save("./output8.png")
```

### System Info

diffusers0.15.0
ubuntu
python3.8

**Comment by patrickvonplaten** [comment]
cc @yiyixuxu @williamberman

**Comment by yiyixuxu** [comment]
@garyhxfang, are you able to use controlnet pipeline with pretrained checkpoints like in this example?
https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/controlnet

I will start with that

**Comment by williamberman** [comment]
Hey @garyhxfang we appreciate the issue :) The issue description is bit vague for us to be of much help. Some things you could do to investigate would be 

1) Use the intermediate logging code in the training script. How are intermediate images during training looking? Are they varying as expected with different conditionings?
2) How are weights changing? Some very basic checks for nan or zero tensors at points where the controlnet connects to the base unet could be helpful.
3) Can you successfully train the example dataset linked in the readme on your hardware setup? Do you get the same issues there?

We appreciate community issues but high quality issue descriptions which do sometimes require some additional leg work from community members help us fix any mistakes we may have made :)

**Comment by irimsky** [comment]
@yiyixuxu @williamberman 
Hello, I met the same problem now. I would provide a detailed description of my situation:
1. I have successfully trained the example dataset `fill50k` in the readme. It works with different conditioning images.
2. Then I created my own dataset by mimicking the format of `fill50k` dataset. It has only 900 512x512 images (for test). Its conditioning images are all binary masks, which representing the highlight pixels in the target images.
3. I trained a new controlnet on my own data with the same training script in the example. Then I found it generated the same  results whatever conditioning images I put in. (other settings like prompt and random seed are same)
4. I checked the nan and inf. it didn't happen. 
5. Previous model was trained on SDv2-1. The pretrained controlnet model on SDv1-5 worked properly.
6. Then I trained my own data on SDv1-5. The same problem.
7. I found that the intermediate validation results turnd out all black. Training on `fill50k`, the intermediate validation results were also all black, though the controlnet model finally worked.

I wondered why this happened.
